### PR TITLE
feature/OO-9965/dotnet-puppet-integration

### PR DESCRIPTION
**Pull request by GalSvirin**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -7,17 +7,40 @@
...
 #
 # [*source*]
-#   Foreign url to fetch Takipi from. This is optional
+#   Optional. Foreign url to fetch Takipi from.
+#
+# [*manage_java*]
+#   Optional. Java agent only. Installs Java package.
+#
+# [*collector_host*]
+#   Required. .Net agent only. IP of remote collector.
```
Comment (chook) [inline manifests/init.pp:19]: if you're working on the Puppet script, it's OK to finally brush it and make it speak "remote collector" first as that's our standard for Java these days anyway.

File: manifests/init.pp [hunk 2]
```diff
@@ -30,57 +53,93 @@
...
+            source    => '/tmp/takipi.deb',
+            provider  => dpkg,
+          }
+        }
+
+        default:  { fail("Unrecognized operating system for Takipi installation") }
+      }
+      ->
+      exec { 'install takipi':
+        cwd       => '/opt/takipi/etc',
```
Comment (chook) [inline manifests/init.pp:109]: i think it's wise to refresh the entry puppet module to only support the agent in a remote collector mode.

Comment (GalSvirin) [reply manifests/init.pp:109]: Will be changed in separate ticket (OO-10091)

File: README.md [hunk 3]
```diff
@@ -17,23 +17,30 @@
...
-Currently supports RPM and DEB-based Linux distributions.
+Java agent currently supports RPM and DEB-based Linux distributions.
+.Net agent currently supports Windows platform.
 
 ## Module Description
 
-Takipi installs a daemon process and a JVM agent library on your servers. These two components work in tandem to log and send data to Takipi's analysis servers. Together they detect events (such as exceptions or logged errors) happening within your app and automatically create the code needed to log and collect the data you'll need to debug these errors in production.
+Takipi installs a daemon process^ and a VM agent library on your servers. These two components work in tandem to log and send data to Takipi's analysis servers. Together they detect events (such as exceptions or logged errors) happening within your app and automatically create the code needed to log and collect the data you'll need to debug these errors in production.
 
+^ Only in java agent.
```
Comment (HenAmar) [inline README.md]: explain the meaning of daemon process and why it is supported only on linux

**Review by HenAmar — APPROVED** [review]

File: manifests/init.pp [hunk 4]
```diff
@@ -30,57 +53,93 @@
...
       }
+        Class['java'] -> Class['takipi']
     }
+    'Windows': {
+      if ($collector_host == undef) or ($collector_port == undef) {
+        fail("Collector host or port not provided")
+      }
+
+      exec { 'remove_overops':
+        command => 'C:\Windows\System32\msiexec.exe /x {6EB3F996-C92A-4D30-BDA6-75A7B20029EE} /qn /norestart',
```
Comment (shimonmagal) [inline manifests/init.pp]: we could store the guid in a variable

File: README.md [hunk 5]
```diff
@@ -17,23 +17,30 @@
...
 ### What takipi affects
 
+Java agent:
 * Installs Java on the node that Takipi is to be installed, handled by `puppetlabs-java` module. (optional)
 * Installs Takipi base package.
 * Adds the Takipi secret key (found under `/opt/takipi/work/secret.key`) and completes package installation.
 
+.Net agent:
+* Removes previous .Net agent.
+* Downloads and installs a msi of the agent.
```
Comment (shimonmagal) [inline README.md]: a msi -> an MSI / an msi

File: manifests/init.pp [hunk 6]
```diff
@@ -30,57 +53,93 @@
   $secret_key  = 'YOUR_KEY',
   $source      = undef,
   $manage_java = true,
+  $collector_host = undef,
+  $collector_port = undef,
+  $windows_temp_dir  = 'C:\users\Administrator\Downloads',
```
Comment (shimonmagal) [inline manifests/init.pp:58]: why this dir? what if user is not administrator?

**Review by shimonmagal — CHANGES_REQUESTED** [review]

File: manifests/init.pp (hunk 6 above)
Comment (HenAmar) [reply manifests/init.pp:58]: @GalSvirin1 we agreed on  windows_temp_dir => 'C:\Windows\Temp', why did you change it ?

**Review by HenAmar — CHANGES_REQUESTED** [review]

File: manifests/init.pp (hunk 6 above)
Comment (GalSvirin) [reply manifests/init.pp:58]: > @GalSvirin1 we agreed on windows_temp_dir => 'C:\Windows\Temp', why did you change it ?

This folder on Windows is protected, we may find ourselves in a permissions problem, BTW, this is not the `%TEMP%` folder (`%TEMP%` -> `C:\Users\%username%\AppData\Local\Temp`)

Comment (GalSvirin) [reply manifests/init.pp:58]: > why this dir? what if user is not administrator?

This is default path of `downloads` folder, use may configure it as described in the example.

**Pull request merged**